addons/l10n_id_efaktur/tests/test_l10n_id_efaktur_download.py

```diff
@@ -0,0 +1,244 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from freezegun import freeze_time
+from odoo import Command
+from odoo.exceptions import UserError, RedirectWarning
+from odoo.tests import tagged
+from odoo.addons.account.tests.common import AccountTestInvoicingHttpCommon
+from .test_l10n_id_efaktur import TestIndonesianEfaktur
+from markupsafe import Markup
+
+
+@tagged('post_install', '-at_install', 'post_install_l10n')
+class TestIndonesianEfakturDownload(TestIndonesianEfaktur, AccountTestInvoicingHttpCommon):
+
+    def test_efaktur_download_multi_company(self):
+        """ Ensure that generating efaktur document for invoices across multi company is not allowed """
+        # Setup company
+        company_2 = self.company_data_2['company']
+        tax_id = self.env['account.tax'].create({"name": "test tax", "type_tax_use": "sale", "amount": 10.0, "price_include": True, "company_id": company_2.id})
+        self.env['l10n_id_efaktur.efaktur.range'].create({'min': "%013d" % 21, 'max': '%013d' % 30, 'company_id': company_2.id})
+
+        # Setup company across 2 companies
+        invoice_1 = self.env["account.move"].create({
+            "move_type": "out_invoice",
+            "partner_id": self.partner_id.id,
+            "invoice_date": "2019-05-01",
+            "date": "2019-05-01",
+            "invoice_line_ids": [
+                Command.create({"name": "line1", "price_unit": 110.0, "tax_ids": self.tax_id.ids}),
+            ],
+            "l10n_id_kode_transaksi": "01",
+            "company_id": self.company.id,
+        })
+        invoice_2 = self.env["account.move"].create({
+            "move_type": "out_invoice",
+            "partner_id": self.partner_id.id,
+            "invoice_date": "2019-05-01",
+            "date": "2019-05-01",
+            "invoice_line_ids": [
+                Command.create({"name": "line1", "price_unit": 110.0, "tax_ids": tax_id.ids}),
+            ],
+            "l10n_id_kode_transaksi": "01",
+            "company_id": company_2.id
+        })
+
+        invoice_1.action_post()
+        invoice_2.action_post()
+
+        with self.assertRaises(UserError):
+            (invoice_1 + invoice_2).download_efaktur()
+
+    @freeze_time('2019-05-01')
+    def test_efaktur_document_download(self):
```

Comment:
This test does too many things, please split it more.

One would test downloading separately
One would test downloading together
One would test re-downloading with the mismatch error etc